### ISSUE TITLE
[Regression] Community navigation breaks contact list

### ISSUE THREAD

**Issue by vit9696** [vector-im/element-web#16999]
<img width="344" alt="Снимок экрана 2021-04-19 в 16 30 29" src="https://user-images.githubusercontent.com/4348897/115244398-94157800-a12c-11eb-8346-2f89c1cf47f3.png">

### Description

Describe here the problem that you are experiencing, or the feature you are requesting.

### Steps to reproduce

- Have at least 2 communities.
- Click on community A in the community bar.
- Click on any contact in the community in the contact list.
- Click on community B in the community bar.

All contacts will then disappear until you click on community B once more as shown on the screenshot. This is not expected and is actually a regression: what should actually happen is a switch to the community B and the contacts from the community B should be shown in the contact list.

Logs being sent: no

### Version information

- **Platform**: desktop
- **OS**: macOS 11.2.3
- **Version**: 1.7.25

**Comment by t3chguy** [comment]
Communities are deprecated and soon to be replaced with Spaces - which are available in labs today.

**Comment by vit9696** [comment]
Sorry, but what are the estimates for this `soon`? From my experience it sounds quite unusual to regress the working functionality in the latest release, force the end-user to upgrade (there is no way to dismiss an update in the client), and then provide no alternative.

Do not you have a policy that requires you to support the feature as long as the functionality is not removed or, at the very least, as long as a fully featured alternative is not provided? Basically like everyone else does?

**Comment by t3chguy** [comment]
I doubt anyone will have time to action fixes for Communities any time soon, maybe platform team but unlikely

Communities are an Element + Synapse specific feature which never made it into the Matrix Spec, it has been in purgatory for a little while now.

> Sorry, but what are the estimates for this soon?

Whenever it is ready, it is waiting on approval in the Matrix spec before it can be used in production

**Comment by anoadragon453** [comment]
I also experience this bug and it makes me sad. I haven't switched to Communities yet (though I probably should).

Anyways, testing with:

* element-desktop: b30956224452643d4944c148abcacb41ab26b7b5
* element-web: 75aef71fe6599f2a35aef648d74194290e82dde3
* matrix-react-sdk: b0af163002e8252d99b6d7075c83aadd91866735
* matrix-js-sdk: b435b582bd4138877225bc60e412c226feb6f9f5

I can reproduce the bug, but going back one commit on matrix-react-sdk to `5d027ffcf477138c5e07ab903034fa577a00b548`, I cannot reproduce the bug.

These leads me to believe that this regression was introduced by https://github.com/matrix-org/matrix-react-sdk/pull/5825. Though I can't really tell you what specific bit caused it.

**Comment by vit9696** [comment]
cc @turt2live

**Comment by turt2live** [comment]
why am I being cc'd?

**Comment by vit9696** [comment]
Because it is apparently your change that caused the regression.

**Comment by turt2live** [comment]
It's worth noting that although it was my change, the team operates off a zero-blame basis. This means that instead of pointing the finger at someone we prefer to point at the change which broke the functionality and get it fixed. Most often this is the same person that wrote the original change, but it doesn't have to be. 

Given the circumstances around the change, our options are effectively:
1. Ignore the problem
2. Fix the problem
3. Revert the PR

Reverting is not a realistic option given it fixes a larger bug. Fixing the problem is possibly doable, though I can't easily reproduce this issue locally so would be writing it off as the implementation sucking and needing replacing. This then leads us to ignoring the problem: as t3chguy already mentioned, we're replacing the whole thing with a new system which is better in most every way - this is much more appealing.

In addition to all of that, we also have informally established a policy where communities are in maintenance mode at best. This means that unless there is a change that renders the entire feature unusable, we're unlikely to fix it as we'd rather spend the development time on writing its replacement. This is of course not ideal when people are trying to use the old stuff when the new stuff isn't out yet, but I personally believe the balance of resources is correct in biasing towards getting the new out ASAP.

The insider knowledge for how close Spaces is to launching also helps determine this bug's fate: we know we're ramping up to a beta, and given the timelines of expecting that to land we are anticipating that this bug can be weathered until a proper launch of Spaces.

As I mentioned, I was having troubles reliably reproducing the issue as well. This tells me that there is a workaround, even if not an ideal one. It seems to be that a bit of patience and clicking around will fix the issue. From a technical perspective, this is because the `GroupStore` pulls data rather than being pushed data, and the room list store is updating the community filter faster than the group store can acquire data. Once the group store stabilizes itself, the room list is able to reliably filter on the communities.

So to recap: it just doesn't make sense to fix this at the moment, regardless of who or what broke it. If you have information to add which demonstrates that this bug makes the room list impossible to use then we can take a second look.

**Comment by bendschs** [comment]
switching back to old version, communities are key to keep order of large contact lists ...

**Comment by t3chguy** [comment]
Closing as communities issues will not be fixed as it is around the corner from being killed off
